server/data/s3Client.ts

```diff
@@ -33,17 +41,52 @@ export default class S3Client {
       Bucket: this.bucket,
       Key: key,
     })
+    const response = await this.s3.send(command)
+    const readableBody = response.Body as Readable
+    const chunks: Uint8Array[] = []
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const chunk of readableBody) {
+      chunks.push(chunk)
+    }
+    return Buffer.concat(chunks).toString('utf8')
+  }
+
+  async putObject(
+    key: string,
+    data: string | Readable | Buffer | Uint8Array,
+    options: Omit<PutObjectCommandInput, 'Bucket' | 'Key' | 'Body'> = {}
+  ): Promise<void> {
+    const command = new PutObjectCommand({
+      Bucket: this.bucket,
+      Key: key,
+      Body: data,
+      ...options,
+    })
+    await this.s3.send(command)
+  }
 
+  async deleteObject(key: string): Promise<void> {
+    const command = new DeleteObjectCommand({
+      Bucket: this.bucket,
+      Key: key,
+    })
+    await this.s3.send(command)
+  }
+
+  async listObjects(prefix?: string): Promise<string[]> {
+    const command = new ListObjectsV2Command({
+      Bucket: this.bucket,
+      Prefix: prefix,
+    })
     const response = await this.s3.send(command)
-    return (await getStream.buffer(response.Body as Readable)).toString()
+    return response.Contents.map(object => object.Key)
   }
 
   async selectObjectContent(s3selectParams: SelectObjectContentCommandInput): Promise<unknown[]> {
     const command = new SelectObjectContentCommand(s3selectParams)
     const response = await this.s3.send(command)
-
     const records: Uint8Array[] = []
-    /* eslint-disable-next-line */
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
I keep bumping into this and I wonder if we could get rid of this rule?

Reply:
me too. i think we can scrap it when we move to node 16, because it allows for more modern features